### PR TITLE
GEOMESA-1745 Kryo Json serialization occasionally causes buffer overf…

### PR DESCRIPTION
**Pull request by aheyne**
…lows

* Now writes out a 0 to the buffer instead of just moving forward in it.

Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

**Comment by aheyne** [comment]
@elahrvivaz Does the start position need to be recorded after the Int is written to out?

File: geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/KryoJsonSerialization.scala [hunk 1]
```diff
@@ -210,7 +210,7 @@ object KryoJsonSerialization extends LazyLogging {
   // write a document without a name - used for the outer-most object which doesn't have a key
   private def writeDocument(out: Output, value: JObject): Unit = {
     val start = out.position()
-    out.setPosition(start + 4) // skip 4 bytes so we have space to go back and write total length
+    out.writeInt(0) // Write int out to prevent oob exceptions so we can resize the buffer at an appropriate time.
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/KryoJsonSerialization.scala]: can you leave the original comment about leaving space to go back and write the length?

**Comment by elahrvivaz** [comment]
nope, start should stay the way you have it.

**Comment by elahrvivaz** [comment]
88d69495bdcb64a8c9c458f197e6f5741e450385

**Pull request closed without merging**